File: backend-java/src/main/java/com/github/pascualex/oc/utils/TeacherConditionalFilters.java
package com.github.pascualex.oc.utils;

import com.github.pascualex.oc.model.Teacher;

import java.util.stream.Stream;

public final class TeacherConditionalFilters {

    public static Stream<Teacher> bySearch(Stream<Teacher> teachers, String search) {
        if (search == null) return teachers;
        String[] words = search.toLowerCase().split("-");
        return teachers.filter(teacher -> {
            String formattedName = teacher.getName().toLowerCase();
            for (String word : words) { if (!formattedName.contains(word)) return false; }
            return true;
        });
    }
}

File: backend-node/src/utils/teacherConditionalFilters.ts
import { Teacher } from '../model/Teacher';

export function bySearch(teachers: Teacher[], search: string): Teacher[] {
  if (!search) return teachers;
  const words: string[] = search.toLowerCase().split('-');
  return teachers.filter((teacher) => {
      const formattedName: string = teacher.name.toLowerCase();
      for (const word of words) { if (!formattedName.includes(word)) return false; }
      return true;
  });
}